### PR TITLE
Use `is not null` syntax

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

* `-replace '!= null',  'is not null'` for C# 9.0.
* Enable C# 9.0 in `xUnit.tests`.

## PR Context

Follow-up #13277.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/powershell/13286)

### CODE REVIEW

**Comment by xtqqczze** [comment]
@

**Comment by iSazonov** [comment]
The pattern comes from C# 9.0 and is already enabled.

**Comment by xtqqczze** [comment]
@iSazonov What is mean is I think we should go for "low-hanging fruit" such as using null conditional access from C# 6.0 ([RCS1146](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1146.md)) etc before making C# 9.0 changes.

**Comment by xtqqczze** [comment]
@iSazonov I have started work on related chanegs in #13286.

**Comment by xtqqczze** [comment]
Build failures seem to be because preview features not enabled in `xUnit.tests`.

https://dev.azure.com/powershell/PowerShell/_build/results?buildId=58427&view=logs&j=ce47e5d9-6379-5300-b411-5c447691f9de&t=2718afa2-89c3-5dee-4660-3339d525f0c9&l=37

```cmd
csharp/test_PSConfiguration.cs(100,47): error CS8652: The feature 'not pattern' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version. [/Users/runner/work/1/s/test/xUnit/xUnit.tests.csproj]
```

**Comment by iSazonov** [comment]
You can exclude xUnit files from the PR or enable C# 9.0 in the csproj.

File: test/Test.Common.props
Comment (iSazonov) [inline test/Test.Common.props:8]:
```suggestion
    <LangVersion>preview</LangVersion>
```

**Comment by xtqqczze** [comment]
@iSazonov If you would like, you can start to review, as we are not isolating C# 9.0 changes. This is already a huge PR so it seems to be the best plan. Any more changes can be in follow-up PR.

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
Macos failures look like `Test-Connection` DNS issues related to #12935.

**Comment by iSazonov** [comment]
@xtqqczze Please rebase to pass CIs.

**Comment by xtqqczze** [comment]
@iSazonov Too many merge conflicts, I will push new commits rather then resolving.

**Comment by xtqqczze** [comment]
First rebase to 1f904e47d

**Comment by xtqqczze** [comment]
Second rebase fixes up.

**Comment by daxian-dbw** [comment]
I don't think the `is not` language feature is for replacing `!=`. There is nothing wrong using `!=`.

**Comment by iSazonov** [comment]
@daxian-dbw What do you mean? The pattern "is not null" is for replacement `!(x is not null)` https://github.com/dotnet/roslyn/issues/42368

**Comment by daxian-dbw** [comment]
Quote from dotnet/roslyn#42368:
> Add feature to convert from `!(x is null)` to `x is not null`

@iSazonov That conversion makes sense. But most changes in this PR are converting `x != null` to `x is not null`, which is not what is suggested. Unlike `!(x is null)`, `x != null` is explicit and clear, it's conventional and there is no readability issue about it.

Even the `!(x is null)` is pretty contrived. IMO, `is not` feature is for cases where we currently use `!(x is <a-type-name>)`.

**Comment by vexx32** [comment]
@daxian-dbw it's worth noting that `==` and `!=` operators can both (potentially) be overridden by a type with an implementation that returns unexpected results for a comparison with null; `x is not null` avoids this possibility.

**Comment by daxian-dbw** [comment]
@vexx32 Yes, there is such possibility, but I don't think it's practically a problem we need to concern about.

**Comment by xtqqczze** [comment]
Just to say that if `x` is a value type, `x != null` could result in undefined behaviour (see #13285 for examples). `x is not null` will result in a [compiler error](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0037) instead .

**Comment by daxian-dbw** [comment]
@xtqqczze I agree that for value types, `x != null` can actually be considered a bug. I will review #13285 shortly.

**Comment by daxian-dbw** [comment]
Given that #13277 is reverted, I will close this PR as well. The idea is the same as in https://github.com/PowerShell/PowerShell/pull/13277#issuecomment-666739038, that we shouldn't force `is not null` over `!= null` -- neither should be universally preferred; they are appropriate in different scenarios.

**Comment by xtqqczze** [comment]
@daxian-dbw I have made `!(x is <a-type-name>)` changes in #13323

**Comment by iSazonov** [comment]
I think we should re-review this again since .Net team added new Roslyn analyzer to force using new syntax "is null" and "is not".

**Comment by xtqqczze** [comment]
We should wait until #13333 is merged to avoid conflicts.

**Pull request closed without merging**
